withdraw-by-check/REPWRITERSPECS/BANNO.CHECK.WITHDRAW.V1.POW

```diff
@@ -1061,5 +1166,387 @@ PROCEDURE BNODEBUGMSGADD
+
+ CALL BNOERRGET
+
+ JSONPROPNAME="errorCode"
+ CALL JSONPROP
+ PRINT FORMAT("999",BNOERRCODE)
+ NEWLINE
+
+ PRINT ","
+ JSONPROPNAME="loggingErrorMessage"
+ CALL JSONPROP
+ PRINT QQ+BNOERRMESSAGE+QQ
+ NEWLINE
+
+ IF BNOERRDISPLAYMSGCOUNT>0 THEN
+  DO
+   PRINT ","
+   JSONPROPNAME="displayErrorMessage"
+   CALL JSONPROP
+
+   CALL JSONARRAY
+   FOR TMPLOOP=1 TO BNOERRDISPLAYMSGCOUNT
+    DO
+     TMPCHR=BNOERRDISPLAYMSG(TMPLOOP)
+     CALL NTS
+     PRINT Q+TMPCHR+Q
+     IF TMPLOOP<BNOERRDISPLAYMSGCOUNT THEN
+      PRINT ","
+     NEWLINE
+    END
+   CALL JSONARRAYEND
+  END
+
+ IF TRANSTATUS=TRANLIMITS AND
+    BNOERRCODE=BNOERRCODEINVALIDAMT THEN
+  DO
+   PRINT ","
+   TMPCHR=FORMAT(" ###,##9.99",RGWDAMOUNT)
+   CALL NLS
+   JSONPROPNAME="requested"
+   CALL JSONPROP
+   PRINT Q+TMPCHR+Q
+   NEWLINE
+
+   PRINT ","
+   TMPCHR=FORMAT(" ###,##9.99",PARAMMINWDLIMIT)
+   CALL NLS
+   JSONPROPNAME="minWdAmount"
+   CALL JSONPROP
+   PRINT Q+TMPCHR+Q
+   NEWLINE
+
+   PRINT ","
+   TMPCHR=FORMAT(" ###,##9.99",PARAMMAXWDLIMIT)
+   CALL NLS
+   JSONPROPNAME="maxWdAmount"
+   CALL JSONPROP
+   PRINT Q+TMPCHR+Q
+   NEWLINE
+  END 
+END [PROCEDURE]
+
+[*
+**  --------------------------------------------------------------------
+**  JSON Procedures - Common
+**  --------------------------------------------------------------------
+*]
+PROCEDURE JSONOPEN
+[* Print JSON open char
+*]
+ PRINT "{"
+ NEWLINE
+END [PROCEDURE]
+
+PROCEDURE JSONCLOSE
+[* Print JSON close char
+*]
+ PRINT "}"
+ NEWLINE
+END [PROCEDURE]
+
+PROCEDURE JSONOBJ
+
+ PRINT "{"
+END [PROCEDURE]
+
+PROCEDURE JSONOBJEND
+
+ PRINT "}"
+ NEWLINE
+END [PROCEDURE]
+
+PROCEDURE JSONARRAY
+
+ PRINT "["
+END [PROCEDURE]
+
+PROCEDURE JSONARRAYEND
+
+ PRINT "]"
+ NEWLINE
+END [PROCEDURE]
+
+PROCEDURE JSONPROP
+
+ PRINT QQ+JSONPROPNAME+QQ+":"
+END [PROCEDURE]
+
+[*
+**  --------------------------------------------------------------------
+**  Utility Procedures
+**  --------------------------------------------------------------------
+*]
+
+PROCEDURE NORMALIZEDATA
+[* Normalize the data and if a parameter, separate into
+** parameter name and value
+** PASS    LFLINE          Character line read in
+** RETURN  PARAMNAME       Parameter name found
+** RETURN  PARAMVALUE      Parameter value
+**
+*]
+ PARAMNAME=""
+ PARAMVALUE=""
+
+ POS=CHARACTERSEARCH(LFLINE,":")
+ IF LFLINE<>"" AND
+    SEGMENT(LFLINE,1,1)<>"*" AND
+    POS>1 THEN
+  DO
+   TMPCHR=SEGMENT(LFLINE,1,POS-1)
+   CALL NLTS
+   PARAMNAME=UPPERCASE(TMPCHR)
+
+   TMPCHR=SEGMENT(LFLINE,POS+1,LENGTH(LFLINE))
+   CALL NLTS
+   PARAMVALUE=TMPCHR
+  END
+END [PROCEDURE]
+
+PROCEDURE NLTS
+[* Remove leading AND trailing spaces
+**  PASS       TMPCHR
+**  RETURN     TMPCHR
+*]
+ CALL NLS
+ IF TMPCHR<>"" THEN
+  CALL NTS
+END [PROCEDURE]
+
+PROCEDURE NLS
+[* Remove leading spaces
+**  PASS       TMPCHR
+**  RETURN     TMPCHR
+*]
+ WHILE (LENGTH(TMPCHR)>0 AND
+        SEGMENT(TMPCHR,1,1)=" ")
+  DO
+   IF LENGTH(TMPCHR)=1 THEN
```

Comment:
I know this is old code.......but as used here... IF LENGTH(TMPCHR)=1 can never be.... since in order to enter the DO/END structure, the SEGMENT(TMPCHR,1,1) must equal=" " and since LENGHT ignores trailing spaces, TMPCHR can never be " " and have a length of 1 at the same time.

Maybe....

```
 WHILE (LENGTH(TMPCHR)>0 AND
        SEGMENT(TMPCHR,1,1)=" ")
  DO
   IF TMPCHR=" " THEN
    TMPCHR=""
   ELSE
```